lib/karafka/connection/messages_processor.rb

```diff
@@ -38,34 +38,22 @@ def process(group_id, kafka_messages)
         # @param controller [Karafka::BaseController] base controller descendant
         # @param mapped_topic [String] mapped topic name
         # @param kafka_messages [Array<Kafka::FetchedMessage>] raw messages from kafka
-        def process_batch(controller, mapped_topic, kafka_messages)
-          messages_batch = kafka_messages.map do |kafka_message|
-            # Since we support topic mapping (for Kafka providers that require namespaces)
-            # we have to overwrite topic with our mapped topic version
-            # @note For the default mapper, it will be the same as topic
-            # @note We have to use instance_variable_set, as the Kafka::FetchedMessage does not
-            #   provide attribute writers
-            kafka_message.instance_variable_set(:'@topic', mapped_topic)
```

Comment:
this remapping needs to happen, otherwise if you remap with staging_ you will get messages with staging_ back, which means that someone might have a business logic relying on the namespaced topic name instead of the proper unmapped to the env one (in your case). I will fix that once it is merged so no worries :)